src/Expander.hs

```diff
@@ -551,52 +558,53 @@ initializeKernel = do
               v <- freshMacroVar
               bind b $ EIncompleteMacro v theName macroDest
               return (theName, v, macroDest)
-            linkDecl dest $ DefineMacros macros
-            nowValidAt pdest (SpecificPhase p)
+            linkOneDecl dest $ DefineMacros macros
+            linkDeclValidity vp (ScopeSet.insertAtPhase p sc scs)
         )
       , ( "example"
-        , \ sc dest pdest stx -> do
+        , \ dest scs vp stx -> do
             p <- currentPhase
             Stx _ _ (_ :: Syntax, expr) <- mustHaveEntries stx
             exprDest <- liftIO $ newSplitCorePtr
             sch <- liftIO newSchemePtr
-            linkDecl dest (Example (view (unSyntax . stxSrcLoc) stx) sch exprDest)
+            linkOneDecl dest (Example (view (unSyntax . stxSrcLoc) stx) sch exprDest)
+            sc <- freshScope $ T.pack $ "For example at " ++ shortShow (stxLoc stx)
```

Comment:
I'm only creating a `Scope` because the code was previously using its input `Scope`. But why does an `example` need a `Scope`? It doesn't bind any new identifier.

Reply:
It doesn't. That was an infelicity in the old approach that you've resolved with this generalization.

Reply:
excellent, removed the `freshScope` call.